### PR TITLE
docs: document supported extensions apis

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
A few folks have asked about what extensions APIs we support. I thought it would make sense to maintain a central list of the "blessed" APIs.

Ref #19447

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by malept** [comment]
Should a link to this be added to `docs/index.md`?

**Comment by sentialx** [comment]
@nornagon Would be great if there would be support for `chrome.webRequest` API, since it's also very tricky to implement in plain JS, because the Electron's `webRequest` module doesn't allow registering multiple handlers (https://github.com/electron/electron/issues/18301).

**Comment by nornagon** [comment]
> Should a link to this be added to `docs/index.md`?

That file doesn't seem to exist..?

**Comment by malept** [comment]
Er, I meant `docs/README.md`.

File: docs/README.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ These individual tutorials expand on topics discussed in the guide above.
 * [Process Object](api/process.md)
 * [Supported Command Line Switches](api/command-line-switches.md)
 * [Environment Variables](api/environment-variables.md)
+* [Chrome Extensions support](api/extensions.md)
```
Comment (malept) [inline docs/README.md]:
```suggestion
* [Chrome Extensions Support](api/extensions.md)
```

File: docs/api/extensions.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# Chrome Extension Support
+
+Electron supports a subset of the [Chrome Extensions
+API][chrome-extensions-api-index], primarily to support DevTools extensions and
+Chromium-internal extensions, but it also happens to support some other
+extension capabilities also.
```
Comment (malept) [inline docs/api/extensions.md]: I do the double-word thing a lot, too.

```suggestion
extension capabilities.
```

File: docs/api/extensions.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+[chrome-extensions-api-index]: https://developer.chrome.com/extensions/api_index
+
+> **Note:** Electron does not support arbitrary Chrome extensions from the
+> store, and it is a **non-goal** of the Electron project to be perfectly
+> compatible with Chrome's implementation of Extensions.
+
+## Loading extensions
+
+Electron only supports loading unpacked extensions (i.e. .crx files do not
```
Comment (malept) [inline docs/api/extensions.md]:
```suggestion
Electron only supports loading unpacked extensions (i.e., `.crx` files do not
```

**Comment by jkleinsc** [comment]
Merging as CI failures unrelated to doc only change.

**Pull request merged**